### PR TITLE
Gh 2270 JUnit5 upgrade (modules: cache, common-utils, data, exception)

### PR DESCRIPTION
**Pull request by macenturalxl1**
Hi - here is the first iteration of the JUnit5 upgrade ticket.

I'll break the root level modules in to manageable batches of work so that it hopefully makes PRs easier to review (as oppose to one large piece). I'll make a note of this on the ticket too.

Here is the first 4x modules under core - I'll raise another PR for the second x4 (graph, operation, serialisation, store) as a part 2.

### CODE REVIEW

File: library/flink-library/src/test/java/uk/gov/gchq/gaffer/flink/operation/FlinkTest.java [hunk 1]
```diff
@@ -122,7 +122,7 @@ public static Store createStore() {
 
     public static void verifyElements(final Graph graph) throws OperationException, InterruptedException {
         // Wait for the elements to be ingested.
-        Thread.sleep(2000);
+        Thread.sleep(3000);
```
Comment (d47853) [inline library/flink-library/src/test/java/uk/gov/gchq/gaffer/flink/operation/FlinkTest.java:125]: Is this failing at the moment?

File: core/common-util/src/test/java/uk/gov/gchq/gaffer/commonutil/JsonAssert.java [hunk 2]
```diff
@@ -17,25 +17,27 @@
...
+
 public final class JsonAssert {
     private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
 
     private JsonAssert() {
         //Private to prevent instantiation
     }
 
-    public static void assertEquals(final String expectedJson, final String actualJson) {
+    public static void assertJsonEquals(final String expectedJson, final String actualJson) {
```
[the remark this answers is not in the document]
Comment (macenturalxl1) [reply core/common-util/src/test/java/uk/gov/gchq/gaffer/commonutil/JsonAssert.java]: Agreed I've reverted this back to original method name

File: core/data/src/test/java/uk/gov/gchq/gaffer/data/element/function/FunctionTest.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
+/*
```
[the remark this answers is not in the document]
Comment (macenturalxl1) [reply core/data/src/test/java/uk/gov/gchq/gaffer/data/element/function/FunctionTest.java]: I've added documentation to this class with a @see annotation to reference the koryphe dependency and why (in regards to JUnit5)

File: library/flink-library/src/test/java/uk/gov/gchq/gaffer/flink/operation/FlinkTest.java (hunk 1 above)
Comment (macenturalxl1) [reply library/flink-library/src/test/java/uk/gov/gchq/gaffer/flink/operation/FlinkTest.java:125]: I think it originally failed once on the pipeline and intermittently on my personal machine (maybe it is slightly slower in some test runs) so the assertion runs too early and fails the test or breaks pipeline

I can change it back to 2000?

Comment (d47853) [reply library/flink-library/src/test/java/uk/gov/gchq/gaffer/flink/operation/FlinkTest.java:125]: No it's fine just might be worth noting if it's unreliable.

**Pull request merged**